### DISCUSSION TITLE
Cannot send reply from onBeforeRequest

### DISCUSSION THREAD

**Discussion by Extarys** [elysiajs/elysia#597] [Q&A]
I have this function that will verify an access token. I cannot seem to find a way to return data from it.

All the examples shows that this should be possible, but only setting the status works. I also tried with `new Response`.
If I set the status but don't return anything, obviously the route continue on it's path and send the reply from the requested url.

```
export const checkAuthn = (app: Elysia) => {
	return app
		.decorate("atokKey", "key")
		.onBeforeHandle(
			{ as: "global" },
			async ({ request: { headers }, atokKey, set, error }) => {
				console.log(`onBeforeHandle`);
				set.status = 418;
				
				return {};
			}
		)
}
```
How can I return a custom object with the status code ?

**Comment by mtt-artis** [comment]
Personally I `throw error(401, "Unauthorized");`

**Reply by Extarys** [reply]
Yes, that is the only thing that seem to return a value to the client. But I find it lack if I want a custom structure like a detailed object of what went wrong.
From the docs, it also seems I should be able to return an object directly and be able to access it from the route handler, but anything other than a `throw error` freezes :\.
